packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/palette/Palette.tsx

```diff
@@ -426,11 +426,17 @@ export const Palette = ({
     hideDiagramElements([diagramElementId], true);
   };
 
-  if (!palette) {
-    return null;
-  }
+  const shouldRender = (): boolean => {
+    if (!palette) {
+      return false;
+    }
+    if (!node && !toolCount) {
+      return false;
+    }
+    return true;
+  };
```

Comment:
No need to create a function for that.  Something like this should be enough:

```
const shouldRender = palette && node && toolCount > 0
if (!shouldRender) {
  return null;
}
```

Reply:
Okay, I removed the function and used ` palette && (node || (!node && toolCount > 0));` to reflect that the palette should be rendered when a node is selected and no tools from the back end are retrieved  (to use fade and hide for example)